src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,137 @@
 import React from 'react';
+import classNames from 'classnames';
+import { getTodos } from '../../api/api';
 import './TodoList.scss';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+type Props = {
+  selectedUserId: number;
+  onSelectedId: (userId: number) => void;
+};
+type State = {
+  userId: number;
+  todos: [] | Todo[];
+  query: string;
+  filterBy: string;
+};
+export class TodoList extends React.Component<Props, State> {
+  state: State = {
+    userId: 0,
+    todos: [],
+    query: '',
+    filterBy: '',
+  };
+
+  async componentDidMount() {
+    const todos = await getTodos();
+
+    this.setState({ todos });
+  }
+
+  handelChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ query: event.target.value });
+  };
+
+  handelSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({ filterBy: event.target.value });
+  };
+
+  prepareTodos = () => {
+    const { todos, query, filterBy } = this.state;
+    let visibleTodos = todos.filter(todo => (
+      todo.title.toLowerCase().includes(query.toLowerCase())
```

Comment:
for query.toLowerCase() better make a separate variable not to calculate it on every iteration